packages/firestore/src/local/persistence_promise.ts

```diff
@@ -224,10 +224,18 @@ export class PersistencePromise<T> {
    * to resolve.
    */
   static forEach<R, S>(
-    collection: { forEach: (cb: (r: R, s?: S) => void) => void },
+    collection: { forEach: (cb: (r: R, s: S) => void) => void },
     f:
       | ((r: R, s: S) => PersistencePromise<void>)
       | ((r: R) => PersistencePromise<void>)
+  ): PersistencePromise<void>;
+  static forEach<R>(
+    collection: { forEach: (cb: (r: R) => void) => void },
+    f: (r: R) => PersistencePromise<void>
+  ): PersistencePromise<void>;
+  static forEach<R, S>(
+    collection: { forEach: (cb: (r: R, s?: S) => void) => void },
+    f: (r: R, s?: S) => PersistencePromise<void>
```

Comment:
I split this into two overloads. One for collections with keys and values, one for collections with just keys.  Without this, you get:

> Argument of type 'S | undefined' is not assignable to parameter of type 'S'.  Type 'undefined' is not assignable to type 'S'.

from the `f.call(this, r, s)` line below (for `s` being undefined).